elytron/WFCORE-3666-alternative-rolemapping.adoc

```diff
@@ -0,0 +1,82 @@
+= WFCORE-3666 Provide Elytron alternative to RoleMappingLoginModule
+:author:            Martin Mazanek
+:email:             mmazanek@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Currently, Elytron role mapping consist mostly of logical and basic add suffix/prefix RoleMappers, which makes it hard to map roles based on an actual map. This task is to introduce more ways of mapping roles, such as using a map or regex, without the customers needing to implement their own custom components.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3666[WFCORE-3666]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ELY-1591[ELY-1591]
+* https://issues.jboss.org/browse/EAP7-1008[EAP7-1008]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Affected Projects or Components
+
+* WildFly, Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* Implement a Map backed RoleMapper for a -> b mapping. There should be options to keep/remove the original mapped role, and to keep/remove roles that weren't mapped. The map would map a role to one or more new roles.
+[source,xml]
+----
+<mappers>
+...
+	<mapped-role-mapper name="my-mapped-role-mapper" removeMapped="true" removeNotMapped="false" />
+		<role-mapping from="roleToMap" to="mappedRole1 mappedRole2" />
```

Comment:
In model this is STRING and items are delimited with comma?
What about using LIST of STRING in model?

Reply:
Model is LIST of STRING. And xml I was thinking STRING_LIST parser.